### PR TITLE
[KOGITO-2351] - Integrate golint with PR checks on kogito-cloud-opera…

### PR DESCRIPTION
**Pull request by Kaitou786**
…tor repository

See: https://issues.redhat.com/browse/KOGITO-2351

Signed-off-by: Tarun Khandelwal <tarkhand@redhat.com>

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ jobs:
       - run: go mod tidy
       - run: go fmt ./...
       - run: go vet ./...
+      - run: go get -u golang.org/x/lint/golint
```
Comment (radtriste) [inline .circleci/config.yml:16]: Can you put that command at the beginning, before the checkout ? 
so we separate node installation and check commands

File: hack/go-lint.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/env bash
+#!/bin/bash
```
Comment (radtriste) [inline hack/go-lint.sh:1]: This was change some time ago to `#!/bin/env bash`
@ricardozanini can you recall why ?

File: .circleci/config.yml (hunk 1 above)
Comment (Kaitou786) [reply .circleci/config.yml]: Done.

File: hack/go-lint.sh (hunk 2 above)
Comment (ricardozanini) [reply hack/go-lint.sh:1]: Ubuntu local clients.

**Comment by radtriste** [comment]
@ricardozanini @Kaitou786 Is there any plan to integrate that `golangci-lint` check into the `hack/go-lint.sh` ?
because `make test` is passing on our machines but then CircleCi may fail ...

**Comment by radtriste** [comment]
https://kie.zulipchat.com/#narrow/stream/232676-kogito/topic/Operator.20and.20golint/near/200070543

**Comment by ricardozanini** [comment]
> @ricardozanini @Kaitou786 Is there any plan to integrate that `golangci-lint` check into the `hack/go-lint.sh` ?
> because `make test` is passing on our machines but then CircleCi may fail ...

Yes :)

**Pull request merged**